src/checker.rs

```diff
@@ -47,17 +47,35 @@ pub fn check(rules: &Rules, raw: &str) -> bool {
         return false;
     }
 
-    let mut words = trimmed.split_whitespace();
+    let words = trimmed.split_whitespace();
     let word_count = words.clone().count();
     if word_count < rules.min_word_count
         || word_count > rules.max_word_count
-        || words.any(|word| rules.disallowed_words.contains(
+        || words.clone().any(|word| rules.disallowed_words.contains(
              &word.trim_matches(|c: char| !c.is_alphabetic()).to_lowercase()
            ))
     {
         return false;
     }
 
+    if !rules.stem_separator_regex.is_empty() && !(rules.stem_separator_regex == "[]") {
+        let regex: Regex = Regex::new(&rules.stem_separator_regex).unwrap();
+        let mut stems_words: Vec<&str> = vec![];
+        
+        for word in words {
+            let maybe_stem_word: &str;
+            maybe_stem_word = &(regex.split(word).nth(0).unwrap_or(word));
```

Comment:
together with the line above:

```suggestion
            let maybe_stem_word = regex.split(word).next().unwrap_or(word);
```

Reply:
Right :) As you know, I had to divide the one-liners to see what is happening in the debugger.

Reply:
Done.